src/Service.Host/client/src/components/ChangeRequests/CreateChangeRequest.js

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+    Page,
+    collectionSelectorHelpers,
+    Dropdown,
+    InputField,
+    itemSelectorHelpers,
+    userSelectors
+} from '@linn-it/linn-form-components-library';
+import Grid from '@mui/material/Grid';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import AssemblyChange from './ChangeTypes/AssemblyChange';
+import employeesActions from '../../actions/employeesActions';
+import changeRequestActions from '../../actions/changeRequestActions';
+import history from '../../history';
+
+function CreateChangeRequest() {
+    const dispatch = useDispatch();
+    useEffect(() => {
+        dispatch(employeesActions.fetch());
+    }, [dispatch]);
+
+    const currentUserId = useSelector(state => userSelectors.getUserNumber(state));
+    const currentUserName = useSelector(state => userSelectors.getName(state));
+
+    const defaultChangeRequest = {
+        enteredBy: {
+            id: parseInt(currentUserId, 10),
+            fullName: currentUserName
+        },
+        proposedBy: {
+            id: parseInt(currentUserId, 10),
+            fullName: currentUserName
+        },
+        changeState: 'PROPOS',
+        changeType: 'PARTEDIT',
+        reasonForChange: '',
+        descriptionOfChange: ''
+    };
+
+    const [item, setItem] = useState(defaultChangeRequest);
+    const employees = useSelector(state => collectionSelectorHelpers.getItems(state.employees));
+
+    const handleFieldChange = (propertyName, newValue) => {
+        if (propertyName === 'proposedBy') {
+            const newProposed = employees.find(e => e.id === parseInt(newValue, 10));
+            setItem(r => ({
+                ...r,
+                proposedBy: { id: newProposed.id, fullName: newProposed.fullName }
+            }));
+        } else {
+            setItem(r => ({ ...r, [propertyName]: newValue }));
+        }
+    };
+
+    const create = () => {
+        dispatch(changeRequestActions.add(item));
+    };
+
+    const createdItem = useSelector(reduxState =>
+        itemSelectorHelpers.getItem(reduxState.changeRequest)
+    );
+
+    useEffect(() => {
+        if (createdItem?.documentNumber) {
```

Comment:
is this effect just to redirect to the newly created thing? Doesn't the itemCreated middleware handle that for us usually?

Reply:
it didn't do it for me so i put this in

Reply:
you might need a rel: 'self' link on your resource

Reply:
yeah put the self link in and i was able to remove this code